### PR TITLE
Safeguard for undefined and fix reference to CHEF_FOUNDATION_VERSION

### PR DESCRIPTION
**Pull request by tpowell-progress**
## Description

Change to `pre-command` is to prevent:

```
C:/bk018dc96f97171dc5b89d/.buildkite/hooks/pre-command: line 19: [: ==: unary operator expected
```

The change in `prep_and_run_tests.ps1` is to match `omnibus_chef_build.ps1` in using `${env:CHEF_FOUNDATION_VERSION}` instead of `$CHEF_FOUNDATION_VERSION`. The environment variable `CHEF_FOUNDATION_VERSION` gets forwarded via `--env CHEF_FOUNDATION_VERSION` in the `docker` invocation, but that will population the `$env:` variable inside the container.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] If `Gemfile.lock` has changed, I have used `--conservative` to do it and included the full output in the Description above.
- [ ] All new and existing tests passed.
- [X] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by tpowell-progress** [comment]
The Windows Unit test run in the verify pipeline should validate the step that is broken in release at the moment.

**Review by sean-simmons-progress — APPROVED** [review]
lgtm

File: .expeditor/scripts/prep_and_run_tests.ps1 [hunk 1]
```diff
@@ -3,7 +3,8 @@ param(
     [Parameter(Position=0)][String]$TestType
 )
 
-. { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -v $CHEF_FOUNDATION_VERSION
+. { Invoke-WebRequest -useb https://omnitruck.chef.io/chef/install.ps1 } | Invoke-Expression; install -channel "current" -project "chef-foundation" -v ${env:CHEF_FOUNDATION_VERSION}
```
Comment (johnmccrae) [inline .expeditor/scripts/prep_and_run_tests.ps1:6]: You don't need the braces for that to evaluate. Correct syntax is `$env:CHEF_FOUNDATION_VERSION`. If you have a thing that needs to evaluate a thing first and drop the answer inline then you would do this: `$($env:CHEF_FOUNDATION_VERSION)`

Comment (tpowell-progress) [reply .expeditor/scripts/prep_and_run_tests.ps1]: Ah, looks like it was just a carryover from `.expeditor/scripts/omnibus_chef_build.ps1`... changing and making the other consistent.

**Review by johnmccrae — APPROVED** [review]

**Pull request merged**